controllers/utils/base_reconciler.go

```diff
@@ -0,0 +1,408 @@
+			// TODO: Filter use owner label to filter. Will need to build generic owner labeling system too.
+			potentiallyOnwedObjs := &unstructured.UnstructuredList{}
+			potentiallyOnwedObjs.SetGroupVersionKind(gvk)
+			err := r.Client.List(r.RequestCtx, potentiallyOnwedObjs)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "encountered when requesting owned objects with gvk %s", gvk)
+			}
+
+			// Filter objects not actually owned by reconciliation subject via owner reference UID.
+			ownedObjs := []unstructured.Unstructured{}
+			for _, pOwned := range potentiallyOnwedObjs.Items {
+				refs := pOwned.GetOwnerReferences()
+				for _, ref := range refs {
+					if ref.UID == r.ReconciliationSubject.GetUID() {
+						ownedObjs = append(ownedObjs, pOwned)
+					}
+				}
+
+			}
+
+			// Check that found objects are ready.
+			for _, owned := range ownedObjs {
+				if ready, found, err := unstructured.NestedBool(owned.Object, "status", "ready"); err != nil {
+					return ctrl.Result{}, errors.Wrapf(err, "encountered when parsing ready for object %s", owned)
+				} else if !found || !ready {
+					if name, found, err := unstructured.NestedString(owned.Object, "metadata", "name"); err != nil {
+						return ctrl.Result{}, errors.Wrapf(err, "encountered when parsing name for object %s", owned)
+					} else if !found {
+						r.Log.Info(fmt.Sprintf("Owned object of kind %s not ready, requeueing", gvk.Kind))
+						return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
+					} else {
+						r.Log.Info(fmt.Sprintf("Owned object %s of kind %s not ready, requeueing", name, gvk.Kind))
+						return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
+					}
+				}
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+}
+
+// RequeueIfCloudStackClusterNotReady requeues the reconciliation request if the CloudStackCluster is not ready.
+func (r *ReconciliationRunner) RequeueIfCloudStackClusterNotReady() (ctrl.Result, error) {
+	if !r.CSCluster.Status.Ready {
+		r.Log.Info("CloudStackCluster not ready. Requeuing.")
+		return ctrl.Result{RequeueAfter: RequeueTimeout}, nil
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupPatcher initializes the patcher with the ReconciliationSubject.
+// This must be done before changes to the ReconciliationSubject for changes to be patched back later.
+func (r *ReconciliationRunner) SetupPatcher() (res ctrl.Result, retErr error) {
+	r.Log.V(1).Info("Setting up patcher.")
+	r.Patcher, retErr = patch.NewHelper(r.ReconciliationSubject, r.Client)
+	return res, errors.Wrapf(retErr, "error encountered while setting up patcher")
+}
+
+// PatchChangesBackToAPI patches changes to the ReconciliationSubject back to the appropriate API.
+func (r *ReconciliationRunner) PatchChangesBackToAPI() (res ctrl.Result, retErr error) {
+	r.Log.V(1).Info("Patching changes back to api.")
+	err := r.Patcher.Patch(r.RequestCtx, r.ReconciliationSubject)
+	return res, errors.Wrapf(err, "error encountered while patching reconciliation subject")
```

Comment:
can we get rid of all the "error encountered while" prefixes? eks-a is driving this kind of simplification in their error messages. Adding prefixes like "error occurred while" is redundant and they stack up to become very messy. It's better to just say "patching reconciliation subject: err" instead of "error encountered while patching reconciliation subject: err"